### PR TITLE
Allow ip range in ip tracking

### PR DESCRIPTION
**Pull request by albertlast**
Allow to search for ip range and
drope the range check for whois provider (get to complex with ipv6...)

### CODE REVIEW

File: Sources/Profile-View.php [hunk 1]
```diff
@@ -1885,33 +1885,38 @@ function TrackIP($memID = 0)
...
 		$context['base_url'] = $scripturl . '?action=profile;area=tracking;sa=ip;u=' . $memID;
 	}
 
 	// Searching?
 	if (isset($_REQUEST['searchip']))
-		$context['ip'] = trim($_REQUEST['searchip']);
+		$context['ip'] = ip2range(trim($_REQUEST['searchip']));
 
-	if (isValidIP($context['ip']) === false)
+	if (count($context['ip']) != 2)
```
Comment (equinoxmatt) [inline Sources/Profile-View.php]: !==

File: Sources/Profile-View.php [hunk 2]
```diff
@@ -1885,33 +1885,38 @@ function TrackIP($memID = 0)
...
+	$ip_string = array('{inet:ip_address_low}','{inet:ip_address_high}');
+	$fields = array(
+			'ip_address_low' => $context['ip']['low'],
+			'ip_address_high' => $context['ip']['high'],
+		);
+
 	$ip_var = $context['ip'];
-	$ip_string = '= {inet:ip_address}';
+
+	if ($context['ip']['low'] !=  $context['ip']['high'])
```
Comment (equinoxmatt) [inline Sources/Profile-View.php]: !==

File: Sources/Profile-View.php (hunk 1 above)
Comment (albertlast) [reply Sources/Profile-View.php]: Can you explain?

File: Sources/Profile-View.php (hunk 2 above)
Comment (albertlast) [reply Sources/Profile-View.php]: Can you explain?

Comment (equinoxmatt) [reply Sources/Profile-View.php]: Sorry, in this context, both of your checks can be done using strict type comparision. Good practice.

File: Sources/Profile-View.php (hunk 1 above)
Comment (equinoxmatt) [reply Sources/Profile-View.php]: See below, but basically you are comparing an integer to an integer, therefore you can use strict type comparision (!==/===) (good practice).

Comment (albertlast) [reply Sources/Profile-View.php]: But which issue solve this?

Comment (equinoxmatt) [reply Sources/Profile-View.php]: It doesn't solve any issue as such. It is good practice to use strict type comparision, unless there is a specific reason you cannot do so.

**Review by equinoxmatt — APPROVED** [review]

**Review by colinschoen — APPROVED** [review]
LGTM

**Pull request merged**